addons/stock/tests/test_report_stock_quantity.py

```diff
@@ -241,3 +241,26 @@ def test_inter_warehouse_transfer(self):
             1.0, 2.0,   # in two days
         ]):
             self.assertEqual(row['product_qty'], qty, "Incorrect qty for Date '%s' Warehouse '%s'" % (row['date:day'], row['warehouse_id'][1]))
+
+    def test_report_stock_quantity_forecast_interval(self):
+        from_date = fields.Date.to_string(fields.Date.add(fields.Date.today(), months=4, days=-1))
+        to_date = fields.Date.to_string(fields.Date.add(fields.Date.today(), months=4, days=4))
+        self.move2.write({'date': fields.Datetime.add(fields.Datetime.now(), months=4), 'date_deadline': fields.Datetime.add(fields.Datetime.now(), months=4)})
+        report = self.env['report.stock.quantity'].read_group(
+            [('date', '>=', from_date), ('date', '<=', to_date), ('product_id', '=', self.product1.id)],
+            ['product_qty', 'date', 'product_id', 'state'],
+            ['date:day', 'product_id', 'state'],
+            lazy=False)
+        forecast_report = [x['product_qty'] for x in report if x['state'] == 'forecast']
+        self.assertEqual(forecast_report, [])
+        self.env['ir.config_parameter'].create({
+            'key': 'stock.forecast_interval_months',
+            'value': 4
+        })
+        report = self.env['report.stock.quantity'].read_group(
```

Comment:
Again, this is a small nitpick, but perhaps some comments here would be a good idea? Like: "# The default interval is 3 months, so there is no forecast" and "# After setting the interval to 4 months we can see a forecast report". But I will not insist on this, the test is short and not that difficult to understand ;)

Reply:
done! ;)